### DISCUSSION TITLE
if config is not present take from a general setting

### DISCUSSION THREAD

**Discussion by cristian-araujo** [spatie/laravel-settings#83] [Q&A]
Is it possible to create global configurations? for example if I have 2 configuration files:
GeneralSettings
RegistrationSettings

I have email configurations in both but if email configuration in registrationsettings is not completed, I need obtain from generalsettings, is this possible?

from RegistrationSettings, something similar to the models:

```php
public function getBannerImageAttribute () {
       return $this->banner ?? app(GeneralSetting::class)->banner;
}
```

Another thing is that this does not work for me and I do not know why

`app (GeneralSetting::class)`

it return

``` php
App\Settings\GeneralSetting {# 653 ▼
   -mapper: Spatie\LaravelSettings\SettingsMapper {# 655 ▶}
   -config: Spatie\ LaravelSettings\SettingsConfig {# 654 ▶}
   -loaded: false
   -configInitialized: true
   #originalValues: null
}
```

**Comment by rubenvanassche** [comment]
At the moment we don't have support for something like the global settings you describe, personally, I would create a repository with both settings classes within the constructor and getters for all the properties that uses the mechanism of your code example with Laravel Models.

As for the problem with `app(GeneralSetting::class)` this is the default behavior, settings only get loaded from the repository when a property is loaded like this:

```php
app(GeneralSetting::class)->site_name
```

Or when you call `toArray` on it:

```php
resolve(GeneralSetting::class)->toArray()
```
